graph/graph.go

```diff
@@ -163,14 +163,11 @@ func (graph *Graph) Register(img *image.Image, layerData archive.ArchiveReader)
 	graph.imageMutex.Lock(img.ID)
 	defer graph.imageMutex.Unlock(img.ID)
 
-	defer func() {
+	var cleanup = func() {
 		// If any error occurs, remove the new dir from the driver.
 		// Don't check for errors since the dir might not have been created.
-		// FIXME: this leaves a possible race condition.
-		if err != nil {
```

Comment:
crazy. I don't have the PoC code from when I was sitting with you at the hackathon. Now I can't recreate it ...
http://play.golang.org/p/3IkKHNN8qE